demos/tests/run_tests.py

```diff
@@ -143,13 +157,29 @@ def option_to_args(key, value):
                     ' '.join(shlex.quote(str(arg)) for arg in case_args))
                 print(flush=True)
 
+                execution_time = -1
+                device = ""
+                pipeline = []
+                for key, value in test_case.options.items():
+                    match = re.search(r'-d.*', key)
+                    if match and device == "":
+                        device = value
```

Comment:
You try to get a single device for each test case, which is IMO misguided. While all current test cases do run all models on the same device, the demos support running different networks on different devices, and I can see us adding test cases that cover that possibility. So it would be more prudent to record (model, device) pairs instead.